utils/duplicates.go

```diff
@@ -24,13 +26,27 @@ import (
 // CheckDuplicates returns error if the provided object slice contains multiple
 // objects sharing the same version/kind/namespace/name combination.
 func CheckDuplicates(objs []*unstructured.Unstructured) error {
-	seen := map[string]struct{}{}
+	seen := map[string]string{}
 	for _, o := range objs {
 		k := fmt.Sprintf("%s, %q, %q", o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName())
-		if _, found := seen[k]; found {
+		v := hash(o)
+		if h, found := seen[k]; found {
+			// allow but elide literal duplicates
+			if h == v {
+				continue
+			}
 			return fmt.Errorf("duplicate resource %s", k)
 		}
-		seen[k] = struct{}{}
+		seen[k] = v
 	}
 	return nil
 }
+
+func hash(obj *unstructured.Unstructured) string {
+	h := sha1.New()
+	// ignore error based on the unvalidated assumption that we already have a valid, marshallable unstructured object
+	json, _ := obj.MarshalJSON()
```

Comment:
I'm totally fine with this, however you could rely on the fact that hash implements a writer and Unstructured implements  custom marshaller interface:

```go
json.NewEncoder(h).Encode(obj)
```

Reply:
PTAL